chain/messagepool/messagepool.go

```diff
@@ -631,6 +632,10 @@ func (mp *MessagePool) checkBalance(m *types.SignedMessage, curTs *types.TipSet)
 		return xerrors.Errorf("failed to check sender balance: %s: %w", err, ErrSoftValidationFailure)
 	}
 
+	if balance == abi.NewTokenAmount(0) {
```

Comment:
FYI, this isn't the right check. It will return true if the fields of the two `abi.TokenAmount` structs (`balance` and `abi.NewTokenAmount(0)`) are the same, which will be false. Use the `IsZero()` method instead.


For more: https://medium.com/learning-the-go-programming-language/comparing-values-in-go-8f7b002e767a.